pynamodb/tests/integration/test_migration.py

```diff
@@ -154,3 +154,105 @@ class Meta:
     BAModel.create_table(read_capacity_units=1, write_capacity_units=1)
     BAModel('pkey', flag=True).save()
     assert (0, 0) == migrate_boolean_attributes(BAModel, ['flag'], allow_rate_limited_scan_without_consumed_capacity=True)
+
+
+@pytest.mark.ddblocal
+def test_legacy_boolean_attribute_deserialization_compatibility(ddb_url):
+    class BAModel(Model):
+        class Meta:
+            table_name = 'lba_deserialization_test'
+            host = ddb_url
+        id = UnicodeAttribute(hash_key=True)
+        flag = BooleanAttribute(null=True)
+        value = UnicodeAttribute(null=True)
+
+    class LBAModel(Model):
+        class Meta:
+            table_name = 'lba_deserialization_test'
+            host = ddb_url
+        id = UnicodeAttribute(hash_key=True)
+        flag = LegacyBooleanAttribute(null=True)
+        value = UnicodeAttribute(null=True)
+
+    BAModel.create_table(read_capacity_units=1, write_capacity_units=1)
+
+    # Create objects with a BooleanAttribute flag
+    BAModel('pkey1', flag=True, value = 'value0').save()
+    BAModel('pkey2', flag=False, value = 'value0').save()
+    BAModel('pkey3', flag=None, value = 'value0').save()
+
+    # Check we are able to read the flag with LegacyBooleanAttribute
+    assert True == LBAModel.get('pkey1').flag
+    assert False == LBAModel.get('pkey2').flag
+    assert None == LBAModel.get('pkey3').flag
+
```

Comment:
Without the change, update_item will throw `TypeError: expected string or buffer`